x/wasm/module.go

```diff
@@ -170,3 +179,35 @@ func (am AppModule) RegisterStoreDecoder(sdr sdk.StoreDecoderRegistry) {
 func (am AppModule) WeightedOperations(simState module.SimulationState) []simtypes.WeightedOperation {
 	return nil
 }
+
+//____________________________________________________________________________
+
+// AddModuleInitFlags implements servertypes.ModuleInitFlags interface.
+func AddModuleInitFlags(startCmd *cobra.Command) {
+	defaults := DefaultWasmConfig()
+	startCmd.Flags().Uint32(flagWasmLRUSize, defaults.LRUCacheSize, "Sets the size in MiB (NOT bytes) of an in-memory cache for wasm contract instances. Set to 0 to disable.")
+	startCmd.Flags().Uint64(flagWasmQueryGasLimit, defaults.SmartQueryGasLimit, "Set the max gas that can be spent on executing a query with a Wasm contract")
+}
+
+// ReadWasmConfig reads the wasm specifig configuration
+func ReadWasmConfig(opts servertypes.AppOptions) (types.WasmConfig, error) {
+	cfg := types.DefaultWasmConfig()
+	var err error
+	if v := opts.Get(flagWasmLRUSize); v != nil {
+		if cfg.LRUCacheSize, err = cast.ToUint32E(v); err != nil {
+			return cfg, err
+		}
+	}
+	if v := opts.Get(flagWasmQueryGasLimit); v != nil {
+		if cfg.SmartQueryGasLimit, err = cast.ToUint64E(v); err != nil {
+			return cfg, err
+		}
+	}
+	// attach contract debugging to global "trace" flag
```

Comment:
Nice one - turn on all debuging metrics with one switch, not a complex series of hidden ones.